models/src/main/scala/coop/rchain/models/BitSetBytesMapper.scala

```diff
@@ -15,7 +15,7 @@ object BitSetBytesMapper {
     )(bitSetToByteString)
 
   def bitSetToByteString(bitset: BitSet): ByteString = {
-    val longs: Array[Long] = bitset.toBitMask
+    val longs: Array[Long] = if (bitset.isEmpty) Array.empty else bitset.toBitMask
```

Comment:
I wonder if there is hysteresis at other boundaries. Would it be better to convert the bitset, and then filter the empties from the back?

Reply:
Good catch! Thanks for that. This fixes the bug '1.' I've mentioned in the commit message.